src/utils/query-actions.js

```diff
@@ -458,6 +458,12 @@ export const queryPromocodes = _.debounce(async (summitId, input, callback, per_
         endpoint.addQuery('filter[]', `code@@${input}`);
     }
 
+    if(extraFilters.hasOwnProperty('class_name')) {
```

Comment:
this should be generic
u should not ask for class_name
any filter that is passed should be sent